gtrending/cli.py

```diff
@@ -0,0 +1,187 @@
+    parent_filter.add_argument(
+        '--language',
+        '--lang',
+        default=""
+    )
+    parent_filter.add_argument(
+        '--since',
+        type=str,
+        choices=['daily', 'monthly', 'weekly'],
+        default='daily'
+    )
+
+
+    parser_repo = subparser.add_parser(
+        'repos',
+        parents=[parent_json, parent_filter]
+    )
+    parser_repo.add_argument(
+        '--spoken-language',
+        '--spoken-lang',
+        default=""
+    )
+    parser_repo.add_argument(
+        '--sort',
+        type=str,
+        choices=['name', 'forks', 'stars'],
+        default='name'
+    )
+    parser_repo.add_argument(
+        '--sort-reverse',
+        type=bool,
+        action=argparse.BooleanOptionalAction,
+        default=False
+    )
+    parser_repo.set_defaults(func=show_repos)
+
+
+    parser_developer = subparser.add_parser(
+        'developers',
+        parents=[parent_json, parent_filter]
+    )
+    parser_developer.set_defaults(func=show_developers)
+
+
+    parser_langs = subparser.add_parser(
+        'langs',
+        parents=[parent_json]
+    )
+    parser_langs.set_defaults(func=show_langs)
+
+
+    parser_spoken_langs = subparser.add_parser(
+        'spoken-langs',
+        parents=[parent_json]
+    )
+    parser_spoken_langs.set_defaults(func=show_spoken_langs)
+
+
+    args = parser.parse_args()
+    if args.command == None:
+        parser.print_help()
+        exit(1)
+    else:
+        args.func(args)
+
+
+def show_developers(args: argparse.Namespace):
+    developers = fetch_developers(
+        language=args.language,
+        since=args.since
+    )
+
+    developers.sort(key = lambda dev: dev['username'])
+
+    if args.json:
+        print_json(developers)
+    else:
+        for dev in developers:
+            print(repr_developer(dev))
+
+
+def show_langs(args: argparse.Namespace):
+    languages = languages_list()
+
+    if args.json:
+        print_json(languages)
+    else:
+        for entry in languages_list():
+            print(entry["name"])
+
+
+def show_spoken_langs(args: argparse.Namespace):
+    spoken_languages = spoken_languages_list()
+
+    if args.json:
+        print_json(spoken_languages)
+    else:
+        for entry in spoken_languages_list():
+            print(entry["urlParam"], entry["name"])
+
+
+def show_repos(args: argparse.Namespace):
+    repos = fetch_repos(
+        language=args.language,
+        spoken_language_code=args.spoken_language,
+        since=args.since
+    )
+
+    repos.sort(
+        key=lambda repo: repo[args.sort],
+        reverse=args.sort_reverse
+    )
+
+    if args.json:
+        print_json(repos)
+    else:
+        for repo in repos:
+            print(repr_repo(repo))
+
+
+def repr_repo(repo: Dict) -> str:
+    title = repo['name']
+    props = {
+        'URL': repo['url'],
+        'Author': repo['author'],
+        'Language': repo['language'],
+        'Forks': repo['forks'],
+        'Stars': repo['stars'],
```

Comment:
(nit:) Are the capitalization a convention or a personal preference? It didn't occur to me that shell scripts likes to have capitalized names for key-value outputs? I can't think of an example of the top of my head (from say, shell utils), though :smile:

Reply:
It can be seen at places.  I personally prefer it because `Capitalization` gives an assurance that the word is referring to the word in `english` language instead of a programming term.

Examples:
`git log`: 
```
commit 1a38b529989b0a4d561028d5bb6f308b742c834d (HEAD -> master, origin/master, origin/HEAD)
Author: Ankit S <ankit.s@postman.com>
Date:   Sat Oct 22 13:58:36 2022 +0530
```

`top`:
```
Processes: 532 total, 4 running, 528 sleeping, 4428 threads                                                       20:26:46
Load Avg: 10.08, 6.15, 5.14  CPU usage: 7.36% user, 18.0% sys, 74.63% idle
SharedLibs: 604M resident, 97M data, 193M linkedit. MemRegions: 704079 total, 1931M resident, 99M private, 2369M shared.
PhysMem: 15G used (2413M wired), 82M unused.
VM: 210T vsize, 6386M framework vsize, 7102827(0) swapins, 8182809(0) swapouts.
```

`curl`:
```
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

```

Reply:
Ah, gotcha. Thanks for the examples